src/compiler/crystal/tools/print_hierarchy.cr

```diff
@@ -3,8 +3,15 @@ require "colorize"
 require "../syntax/ast"
 
 module Crystal
-  def self.print_hierarchy(program, exp)
-    HierarchyPrinter.new(program, exp).execute
+  def self.print_hierarchy(program, exp, format)
+    case format
```

Comment:
This check should be done before this point, before typing (compiling) the program. Otherwise you get the error message a few seconds after you run the tool. For example try executing this:

```
bin/crystal tool h src/compiler/crystal.cr -f foo
```

you will see you only get the error message after a second or two. Ideally this should be checked when you get the value in OptionParser.